src/SDKs/Azure.Core/data-plane/Azure.Core/ReliableStream.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+
+namespace Azure.Core
+{
+    public static class ReliableStream
```

Comment:
This appears to be a bit of a different pattern than used elsewhere for factory/singletons.   Any reason for going with a nested `private` class rather than having a `private` constructor and static factory method?

_(Not that I mean to imply that I see a problem, I'm just being curious as to the thoughts behind the choice.)_

Reply:
2 reasons:
1. ReadOnlyStream is not public so I can't inherit from it in a public class
2. We need to do an async operation in the factory (get a first response) so having just a ctor doesn't work.